ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -34,13 +39,15 @@
 class Log implements Write {
   private final String dataDir;
   private final long capacityInBytes;
+  private final boolean isLogSegmented;
   private final StoreMetrics metrics;
+  private final Iterator<Pair<String, String>> segmentNameAndFileNameIterator;
```

Comment:
The iterator provides the names for any new log segments that are created - it is very different from `segmentsByName` which contains the log segments that are currently part of the log.

The doc on the second constructor should give you an idea about how this is intended to work. Also look at `getNextSegmentNameAndFilename()`.